### PR TITLE
[Feature]Support query paimon system table

### PR DESCRIPTION
**Pull request by leoyy0316**
Why I'm doing:
now, starrocks not support query paimon system table
What I'm doing:
Support query paimon system table through jni connector 
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java [hunk 1]
```diff
@@ -244,8 +249,12 @@ public Statistics getTableStatistics(OptimizerContext session,
...
         long rowCount = 0;
         for (Split split : splits) {
-            DataSplit dataSplit = (DataSplit) split;
-            rowCount += dataSplit.dataFiles().stream().map(DataFileMeta::rowCount).reduce(0L, Long::sum);
+            if (split instanceof DataSplit) {
+                DataSplit dataSplit = (DataSplit) split;
+                rowCount += dataSplit.dataFiles().stream().map(DataFileMeta::rowCount).reduce(0L, Long::sum);
+            } else {
+                rowCount += split.rowCount();
+            }
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java:257]: use `rowCount += split.rowCount()` directly

File: fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java [hunk 2]
```diff
@@ -179,6 +180,10 @@ public Table getTable(String dbName, String tblName) {
             LOG.error("Get paimon table {}.{} createtime failed, error: {}", dbName, tblName, e);
         }
         PaimonTable table = new PaimonTable(this.catalogName, dbName, tblName, fullSchema, paimonNativeTable, createTime);
+        if (!(paimonNativeTable instanceof AbstractFileStoreTable ||
+                paimonNativeTable instanceof ReadOptimizedTable)) {
+            table.setSystemTable(true);
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/connector/paimon/PaimonMetadata.java:185]: refer to the `AbstractCatalog::isSpecifiedSystemTable` in the Paimon project to set this variable

File: fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java [hunk 3]
```diff
@@ -108,6 +109,14 @@ public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength)
         return scanRangeLocationsList;
     }
 
+    public long getEstimatedLength(long rowcount, TupleDescriptor tupleDescriptor) {
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:112]: rowCount

File: fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java [hunk 4]
```diff
@@ -108,6 +109,14 @@ public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength)
...
+        List<Column> dataColumns = tupleDescriptor.getSlots().stream().map(s -> s.getColumn())
+                .collect(Collectors.toList());
+        long presentRowNums = dataColumns.stream().mapToInt(column -> column.getType().getTypeSize()).sum();
```
Comment (miomiocat) [inline fe/fe-core/src/main/java/com/starrocks/planner/PaimonScanNode.java:115]: rowSize

**Review by miomiocat — DISMISSED** [review]

**Review by stephen-shelby — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/PaimonTable.java [hunk 5]
```diff
@@ -35,18 +35,17 @@ public class PaimonTable extends Table {
     private final String catalogName;
     private final String databaseName;
     private final String tableName;
-    private final AbstractFileStoreTable paimonNativeTable;
+    private final org.apache.paimon.table.Table paimonNativeTable;
```
Comment (stephen-shelby) [inline fe/fe-core/src/main/java/com/starrocks/catalog/PaimonTable.java:38]: Why not import statically org.apache.paimon.table.Table?

**Review by packy92 — DISMISSED** [review]

**Review by HangyuanLiu — DISMISSED** [review]

**Review by stephen-shelby — APPROVED** [review]

**Review by miomiocat — APPROVED** [review]

**Review by packy92 — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by miomiocat** [comment]
@mergify backport branch-3.2

**Pull request merged**